### PR TITLE
CVSB-3910 Review screens for multiple vehicles

### PR DESCRIPTION
**Pull request by agoncearenco**

### CODE REVIEW

File: src/pages/testing/test-creation/test-review/test-review.ts [hunk 1]
```diff
@@ -248,7 +275,26 @@ export class TestReviewPage implements OnInit {
     return this.commonFunctions.getCountryStringToBeDisplayed(vehicle);
   }
 
+  /**
+   * Go to overview only if it's the first vehicle in the test, otherwise to previous vehicle review
+   */
   backToTestOverview() {
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/test-review/test-review.ts]: remove this method if it isn't used anymore

File: src/pages/testing/test-creation/test-review/test-review.ts [hunk 2]
```diff
@@ -248,7 +275,26 @@ export class TestReviewPage implements OnInit {
...
     this.navCtrl.pop();
   }
+
+  get nextButtonText(): string {
+    switch (this.latestTest.vehicles.length) {
+      case 1:
+        return 'Submit test';
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/test-review/test-review.ts]: move strings to app.enums please

File: src/pages/testing/test-creation/test-types-list/test-types-list.ts [hunk 3]
```diff
@@ -83,6 +83,7 @@ export class TestTypesListPage implements OnInit {
           test.testTypeCategoryName = this.testTypeCategoryName;
           this.vehicleService.addTestType(vehicleData, test);
           this.navCtrl.popTo(views[i]);
+          break;
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/test-types-list/test-types-list.ts]: what this 'break' does?

File: src/pages/testing/test-creation/test-create/test-create.html [hunk 4]
```diff
@@ -7,7 +7,7 @@
       <button ion-button
               [ngClass]="{buttonOpacity: changeOpacity}"
               (click)="reviewTest()">
-        Review
+        Review and submit
```
[the remark this answers is not in the document]
Comment (agoncearenco) [reply src/pages/testing/test-creation/test-create/test-create.html]: Ok, I will remove it. I had this change specified on the ticket.

File: src/pages/testing/test-creation/test-review/test-review.ts [hunk 5]
```diff
@@ -91,10 +97,8 @@ export class TestReviewPage implements OnInit {
       this.isTestSubmitted = localStorage.getItem(LOCAL_STORAGE.IS_TEST_SUBMITTED);
       this.isTestSubmitted ? this.viewCtrl.showBackButton(false) : this.viewCtrl.showBackButton(true);
     });
-  }
-
-  ionViewWillEnter() {
-    this.viewCtrl.setBackButtonText(APP_STRINGS.TEST);
+    
+    this.viewCtrl.setBackButtonText(this.navParams.get('backButtonText') || APP_STRINGS.TEST);
```
[the remark this answers is not in the document]
Comment (agoncearenco) [reply src/pages/testing/test-creation/test-review/test-review.ts]: Why is this necessary? I use the backButtonText only on the back button. Don't think I will need to use that variable somewhere else.

File: src/pages/testing/test-creation/test-types-list/test-types-list.ts (hunk 3 above)
Comment (agoncearenco) [reply src/pages/testing/test-creation/test-types-list/test-types-list.ts]: This method adds a test type to an existing vehicle and goes back to the TEST_CREATE_PAGE. The action of going back is done with a loop on **all** the views present in navCtrl. If we have multiple views with the TEST_CREATE_PAGE in navCtrl (when you click Change details of a vehicle in the test review pages), this method will go back twice and add the same test type twice. This break will stop the loop at the first TEST_CREATE_PAGE found in navCtrl.

File: src/pages/testing/test-creation/test-create/test-create.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (agoncearenco) [reply src/pages/testing/test-creation/test-create/test-create.html]: Requested change completed.

File: src/pages/testing/test-creation/test-review/test-review.spec.ts [hunk 6]
```diff
@@ -113,4 +123,34 @@ describe('Component: TestReviewPage', () => {
...
+    component.latestTest = newTest;
+    component.vehicleBeingReviewed = component.latestTest.vehicles.length-1;
+
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+      let submitButton = fixture.debugElement.query(By.css('.footer-cta-section>button'));
+      expect(component.nextButtonText).toBe("Submit tests");
+      submitButton.nativeElement.dispatchEvent(new Event('click'));
+      expect(component.submitTest).toHaveBeenCalled();
+      console.log(submitButton.nativeElement);
```
[the remark this answers is not in the document]
Comment (agoncearenco) [reply src/pages/testing/test-creation/test-review/test-review.spec.ts]: Requested change completed.

File: src/pages/testing/test-creation/test-review/test-review.ts [hunk 7]
```diff
@@ -144,7 +151,24 @@ export class TestReviewPage implements OnInit {
...
   }
 
-  submitTest(test: TestModel) {
+  /**
+   * Handler for the next button
+   * Go to next vehicle if there are vehicles left, otherwise submit test
+   */
+  goToNextPage(){
+    if(this.vehicleBeingReviewed < this.latestTest.vehicles.length-1) 
+      this.navCtrl.push(PAGE_NAMES.TEST_REVIEW_PAGE, {vehicleBeingReviewed: <number> this.vehicleBeingReviewed+1, backButtonText: <string> this.title});
```
[the remark this answers is not in the document]
Comment (agoncearenco) [reply src/pages/testing/test-creation/test-review/test-review.ts]: Requested change completed.

File: src/pages/testing/test-creation/test-review/test-review.ts (hunk 1 above)
Comment (agoncearenco) [reply src/pages/testing/test-creation/test-review/test-review.ts]: Requested change completed.

File: src/pages/testing/test-creation/test-review/test-review.ts (hunk 2 above)
Comment (agoncearenco) [reply src/pages/testing/test-creation/test-review/test-review.ts]: Requested change completed.

**Pull request merged**